content/docs/15.how-to-guides/etl-pipelines.md

```diff
@@ -0,0 +1,233 @@
+    region: "eu-central-1"
+    from: "{{ outputs.get_orders_per_product_csv.uri }}"
+    bucket: "my_bucket"
+    key: "orders/orders_per_product" 
+```
+
+Similar Query tasks can be perfomed on different databases like Snowflake, Postgres, etc.
+
+## Using Python
+
+You can choose to perform ETL using python (pandas) and then run it as a Python script.
+
+The ETL performed using [DuckDB](#using-duckdb) above can be performed using Python as shown in the example flow below.
+
+```yaml
+id: python_etl
+namespace: company.team
+
+tasks:
+  - id: etl
+    type: io.kestra.plugin.scripts.python.Script
+    description: Python ETL Script
+    beforeCommands:
+      - pip install requests pandas
+    script: |
+      import io
+      import requests
+      import pandas as pd
+
+      def _extract(url):
+        csv_data = requests.get(url).content
+        return pd.read_csv(io.StringIO(csv_data.decode('utf-8')), header=0)
+      
+      def run_etl():
+        orders_data = _extract("https://huggingface.co/datasets/kestra/datasets/raw/main/csv/orders.csv")
+        products_data = _extract("https://huggingface.co/datasets/kestra/datasets/raw/main/csv/products.csv")
+        # perform join transformation
+        detailed_orders = orders_data.merge(products_data, how='left', left_on='product_id', right_on='product_id')
+        detailed_orders.to_csv("detailed_orders.csv")
+        # perform aggregation
+        orders_per_product = detailed_orders.groupby('product_id').agg(order_count= ('product_id', 'count'), product_count=('quantity', 'sum'), order_total=('total', 'sum')).sort_values('product_id')
+        orders_per_product['order_total'] = orders_per_product['order_total'].apply(lambda x: float("{:.2f}".format(x)))
+        orders_per_product.to_csv("orders_per_product.csv")
+      
+      if __name__ == "__main__":
+          run_etl()
+    outputFiles:
+      - detailed_orders.csv
+      - orders_per_product.csv
+
+  - id: upload_detailed_orders_to_s3
+    type: io.kestra.plugin.aws.s3.Upload
+    description: Upload the resulting CSV file onto S3
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY_ID') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY_ID') }}"
+    region: "eu-central-1"
+    from: "{{ outputs.python_etl.outputFiles('detailed_orders.csv') }}"
+    bucket: "my_bucket"
+    key: "orders/detailed_orders"
+
+  - id: upload_orders_per_product_to_s3
+    type: io.kestra.plugin.aws.s3.Upload
+    description: Upload the resulting CSV file onto S3
+    accessKeyId: "{{ secret('AWS_ACCESS_KEY_ID') }}"
+    secretKeyId: "{{ secret('AWS_SECRET_KEY_ID') }}"
+    region: "eu-central-1"
+    from: "{{ outputs.python_etl.outputFiles('orders_per_product.csv') }}"
+    bucket: "my_bucket"
+    key: "orders/orders_per_product"
+```
+
+## Using Batch Task Runners
+
+When the python scripts get more compute-intesive or memory-intensive, it is advised to run them on remote batch compute resources using Batch Task Runners.
```

Comment:
It's worth noting that the Docker task runner is remote. They still run locally, just in a container. You should change this example to use AWS Batch Runners or equivalent